### DISCUSSION TITLE
Generate a sql file for initializing a db container

### DISCUSSION THREAD

**Discussion by andyli** [prisma/prisma#12179] [Q&A] [answered]
Many docker official database images (e.g. [mysql](https://hub.docker.com/_/mysql) and [postgres](https://hub.docker.com/_/postgres)) allow initializing using sql files mounted at `/docker-entrypoint-initdb.d`. Is there a way to generate such sql file from `schema.prisma`?

**Comment by austincrim** [comment]
`prisma migrate dev --create-only` will generate a `.sql` script you could place in the `initdb` folder.

You can read more [in our docs](https://www.prisma.io/docs/concepts/components/prisma-migrate#customizing-migrations). Let me know if you have more questions!

**Reply by andyli** [reply]
If I understand correctly, the command `prisma migrate dev --create-only` will only generate such `CREATE TABLE` sql file when it run against an empty, running, DB.

Also, there is no way to specify the output file location.

It would be nice if there is a prisma generator that just generate the `CREATE TABLE` init sql file.

**Reply by pimeys** [reply]
You could try it with [migrate diff](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff). You define both sides and it gives you the SQL as a response (or simplified list of changes). `from` is the starting point and `to` is where you want to go.

**Reply by tomhoule** [reply, marked as answer]
> You define both sides and it gives you the SQL as a response (or simplified list of changes). from is the starting point and to is where you want to go.

`from-empty` and `to-schema-datamodel` in this case.

**Comment by AmirL** [comment]
Can be done with command:
```bash
npx prisma migrate diff --from-empty --to-schema-datasource prisma/schema.prisma --script > prisma/generated.sql
```

**Reply by wernert** [reply]
`npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/generated.sql`
works

**Reply by nielsdebruin** [reply]
I found it more stable to include the `-o myfile` flag (output file) instead of piping it to a file, as this might also add other output generated by the npx command to the SQL file.

`npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script -o prisma/generated.sql`

**Comment by zloom** [comment]
Here is typescript tool https://github.com/zloom/prisma-sql-gen

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by Tsanas** [comment]
Pre 7.0, the following worked.:
`npx prisma migrate diff --from-empty --to-schema-datamodel prisma/schema.prisma --script -o init.sql`

But now the following error is given.:
**`--to-schema-datamodel` was removed. Please use `--[from/to]-schema` instead.**

And the help even describes.:
`--to-schema            Path to a Prisma schema file, uses the datamodel for the diff`

Trying to then run this.:
`npx prisma migrate diff --from-empty --to-schema prisma/schema.prisma --script -o init.sql`

Produces no error, but also not an SQL file
